src/frontends/onnx/tests/runtime.py

```diff
@@ -95,6 +109,24 @@ def convert_to_tensors(self, input_values):
             if parameter.get_output_element_type(0) == Type.bf16:
                 input_tensors.append(Tensor(Type.bf16, input_val.shape))
                 input_tensors[-1].data[:] = input_val.view(np.float16)
+            elif parameter.get_output_element_type(0) == Type.f8e5m2:
+                input_tensors.append(Tensor(Type.f8e5m2, input_val.shape))
+                input_tensors[-1].data[:] = np.frombuffer(input_val.astype(dtype=np.float16).tobytes()[1::2], dtype=np.uint8).reshape(input_val.shape)
+            elif parameter.get_output_element_type(0) == Type.f8e4m3:
+                input_tensors.append(Tensor(Type.f8e4m3, input_val.shape))
+                data_f16 = input_val.astype(dtype=np.float16).tobytes()
+                data_f8 = bytearray()
+                for idx in range(0, len(data_f16)//2):
+                  hb = data_f16[idx*2+1]
+                  lb = data_f16[idx*2]
+                  val = (hb << 8) | lb
+                  # detecting NaN and -NaN
+                  if val == 0x7E00 or val == 0xFE00:
+                    val = 0x7F | (hb & 0x80)
+                  else:
+                    val = (((val << 1) & 0x3FFF) >> 8) | (hb & 0xC0)
+                  data_f8.append(val)
+                input_tensors[-1].data[:] = np.frombuffer(data_f8, dtype=np.uint8).reshape(input_val.shape)
```

Comment:
I'm not convinced that such conversion should be a part of the test logic. 
Does it mean that user would need to apply the same trick to make it work?
It may lead to a higher tests pass rate, but hide not supported cases.

Should the related onnx tests with f8 input be enabled within this PR scope?

Reply:
Numpy doesn't have a float8 types at the moment. To be compatible, ONNX operates outside with something existing: float32/16.
Fortunately, fp8 uses mostly internally for a weights compression, so, users don't need to pass a fp8 values on input. It is exactly test-logic to be able pass/receive fp8 to/from a simple model.
And yes, if some user want to pass fp8 to a model - it should be somehow prepared/extracted from somewhere, it isn't related to this code.

Regarding onnx tests - I would be happy to enable them, but we have a very limited support in a plugins. Only Template plugin supports an FP8 operations, even CPU plugin says it cannot convert it.

Here is same situation like around string/int4 enabling - this PR mostly allows to operate with a type somewhere else (fakequantize, extensions, etc) than enables a missing functionality in OpenVINO.

Reply:
I'll try to replace by onnx helpers if it is better